README.md

```diff
@@ -1,5 +1,127 @@
+
+To run a local build:
+  ```
+  $ gulp localBuild
+  $ cd build/default
+  $ npm serve
+  ```
+
+### Running a production build
+You should never run the `gulp prodBuild` task manually. This task is run on Travis as part of our CI/CD process when a change is pushed to production.
+
+## Build a new component
+
+### Developing the component
+1. Generate a basic component structure using the [generator-px-comp](https://github.com/predixdev/generator-px-comp) Yeoman generator.
+2. Write the code to create a new component in that folder. (Have fun! 🎉)
+3. Write tests for your component that are now passing. (Nice work! 💯)
+4. Publish your component on bower:
+	```
+	$ bower register px-COMPONENT-NAME https://github.com/PredixDev/px-COMPONENT-NAME.git
+	```
+
+### Adding interactive documentation
+
+All components should have an interactive documentation and demo page on the [Predix UI catalog](https://www.predix-ui.com). These documentation pages are built automatically by Travis CI for each component when a new release (i.e. a new tag) is pushed to Github.
+
+Before you begin, you'll need to install the following tools:
+
+1. [Ruby](https://www.ruby-lang.org/en/) version 1.9.3 or higher (this could be tricky, and the ruby build that shipped with your system might be too old)
+2. [Travis CLI ruby gem](https://github.com/travis-ci/travis.rb)
+
+You should also do the following before starting:
+
+1. Make sure you switch off of BLUESSO (or any proxied Wifi connection) onto Internet (or any open Wifi connection). You should also turn off your VPN, if you have one.
+2. Disable your proxy settings in your shell (i.e. don't set a http_proxy or https_proxy).
+
+Once you are ready you can:
+1. [Enable the repository](https://docs.travis-ci.com/user/getting-started#To-get-started-with-Travis-CI%3A) you'd like to build in Travis CI. **Note: you only need to follow steps 1 & 2 of the Travis instructions.**
+2. Run the following command, replacing "xxxxx" with your Sauce Labs username and key:
+	```
+	$ travis encrypt "SAUCE_USERNAME=xxxxx SAUCE_ACCESS_KEY=xxxxx" --add
+	```
+	(To find your Sauce Labs key, log into your account > select your name in the top right > select my account > click the "show" button under Access Key)
+3. Check the `.travis.yml` file with your newly encrypted credentials into git.
+
+### Adding the component to site nav and component gallery
+
+1. Add the component to `elements/px-catalog/pages.json`.
+    ```
+    {
+      "path": "px-COMPONENT-NAME",
+      "label": "COMPONENT NAME",
+      "module": "px-COMPONENT-NAME-demo",
+      "entrypoint": "/bower_components/px-COMPONENT-NAME/demo/px-COMPONENT-NAME-demo.html",
+      "redirects": [
```

Comment:
@davidrleonard - this is only for existing components, right? New stuff doesn't need this?

Reply:
Yep redirects only for existing i think.